src/state/transactions/updater.tsx

```diff
@@ -74,57 +75,54 @@ export default function Updater(): null {
   useEffect(() => {
     if (!chainId || !library || !lastBlockNumber) return
 
-    const cancels = Object.keys(transactions)
-      .filter(hash => shouldCheck(lastBlockNumber, transactions[hash]))
-      .map(hash => {
-        const { promise, cancel } = getReceipt(hash)
-        promise
-          .then(receipt => {
-            if (receipt) {
-              dispatch(
-                finalizeTransaction({
-                  chainId,
-                  hash,
-                  receipt: {
-                    blockHash: receipt.blockHash,
-                    blockNumber: receipt.blockNumber,
-                    contractAddress: receipt.contractAddress,
-                    from: receipt.from,
-                    status: receipt.status,
-                    to: receipt.to,
-                    transactionHash: receipt.transactionHash,
-                    transactionIndex: receipt.transactionIndex,
-                  },
-                })
-              )
-
-              addPopup({
-                txn: {
-                  hash,
-                  success: receipt.status === 1,
-                  summary: transactions[hash]?.summary,
```

Comment:
Correct me if I'm wrong but I'd expect the filter to return only one transaction.
If this is the case, having an array with 1 element and map on it is over-engineering.
Nevertheless, I also want to make sure this is OK to proceed.

Reply:
There are no rules in the dapp to prevent the user from submitting two transactions. Here's a one:

Tx\#1: I swap ETH to USDC while waiting for the swap to go through, 
Tx\#2: I switch my wallet address to another and go to the Liquidity page to add liquidity to the XZ-YZ pair. I submit another transaction.

Now I have two transactions. Now my Tx#1 might block checking for Tx#2. I think getting the array list is safer. :)

Reply:
Reverted

Reply:
I was wrong.
Reverted.